packages/core/content-manager/admin/src/preview/pages/Preview.tsx

```diff
@@ -0,0 +1,197 @@
+import * as React from 'react';
+
+import { Page, useQueryParams, useRBAC, createContext } from '@strapi/admin/strapi-admin';
+import { Box, FocusTrap, Portal, Typography } from '@strapi/design-system';
+import { useIntl } from 'react-intl';
+import { useParams } from 'react-router-dom';
+
+import { GetPreviewUrl } from '../../../../shared/contracts/preview';
+import { COLLECTION_TYPES } from '../../constants/collections';
+import { DocumentRBAC } from '../../features/DocumentRBAC';
+import { type UseDocument, useDocument } from '../../hooks/useDocument';
+import { useDocumentLayout } from '../../hooks/useDocumentLayout';
+import { buildValidParams } from '../../utils/api';
+import { useGetPreviewUrlQuery } from '../services/preview';
+
+import type { UID } from '@strapi/types';
+
+/* -------------------------------------------------------------------------------------------------
+ * PreviewProvider
+ * -----------------------------------------------------------------------------------------------*/
+
+interface PreviewContextValue {
+  url: string;
+  mainField: string;
+  document: NonNullable<ReturnType<UseDocument>['document']>;
+  meta: NonNullable<ReturnType<UseDocument>['meta']>;
+  schema: NonNullable<ReturnType<UseDocument>['schema']>;
+}
+
+const [PreviewProvider, usePreviewContext] = createContext<PreviewContextValue>('PreviewPage');
+
+/* -------------------------------------------------------------------------------------------------
+ * PreviewPage
+ * -----------------------------------------------------------------------------------------------*/
+
+const PreviewPage = () => {
+  const { formatMessage } = useIntl();
+
+  // Read all the necessary data from the URL to find the right preview URL
+  const {
+    slug: model,
+    id: documentId,
+    collectionType,
+  } = useParams<{
+    slug: UID.ContentType;
+    id: string;
+    collectionType: string;
+  }>();
+  const [{ query }] = useQueryParams<{
+    plugins?: Record<string, unknown>;
+  }>();
+  const params = React.useMemo(() => buildValidParams(query), [query]);
+
+  if (!collectionType) {
+    throw new Error('Could not find collectionType in url params');
+  }
+
+  if (!model) {
+    throw new Error('Could not find model in url params');
+  }
+
+  // Only collection types must have a documentId
+  if (collectionType === COLLECTION_TYPES && !documentId) {
+    throw new Error('Could not find documentId in url params');
+  }
```

Comment:
Isn't impossible to access this route without these values? I know we want to ensure we have a value, but I'm not sure these errors ever throw since if you don't have these params then I guess you are on another route...right?

Reply:
yes they should never throw, it's just for type safety because everything from useParams is `string | undefined`. I copied the approach from `useDoc`, this way I don't have to use `!`

Reply:
Also with throw you can keep using hooks below, which is not the case when returning something as it breaks the rules of hooks